program/src/instructions/legacy/withdraw_nft.rs

```diff
@@ -67,13 +57,11 @@ pub struct WithdrawNft<'info> {
         seeds=[
             b"nft_receipt".as_ref(),
             mint.key().as_ref(),
-            pool.key().as_ref(),
+            transfer.pool.key().as_ref(),
         ],
         bump = nft_receipt.bump,
         //can't withdraw an NFT that's associated with a different pool
         has_one = mint @ ErrorCode::WrongMint,
-        has_one = pool @ ErrorCode::WrongPool,
-        close = owner,
     )]
     pub nft_receipt: Box<Account<'info, NftDepositReceipt>>,
```

Comment:
How come you removed `close = owner` for this?

Reply:
Anchor macro won't parse sub fields so added manual closing of nft receipts at the end of relevant handlers.